dependencies.yaml

```diff
@@ -229,6 +229,7 @@ dependencies:
           - dask-cuda==24.10.*,>=0.0.0a0
           - joblib>=0.11
           - numba>=0.57
+          - numpy>=1.23,<3.0a0
```

Comment:
It seems we have a NumPy dependency

https://github.com/rapidsai/cuml/blob/973a65ff703a14d9dd8f355af373e2053444b27a/python/cuml/cuml/linear_model/logistic_regression_mg.pyx#L27

However it isn't getting declared as one. So explicitly added NumPy as a dependency